### PR TITLE
Feature/add filter organization units

### PR DESCRIPTION
**Pull request by Hamilcar31**
Related to https://github.com/linagora/linid-im-front-community-plugins/issues/154

### CODE REVIEW

File: tests/features/front/AccountsPage.feature [hunk 1]
```diff
@@ -109,3 +110,22 @@ Feature: Test Account homepage display
     And I expect the HTML element '[data-cy="cell-lastname"]' contains "user2_ln"
     And I expect the HTML element '[data-cy="cell-email"]' contains "user2@example.com"
     And I expect the HTML element '[data-cy="cell-createdBy"]' contains "admin_fn admin_ln"
+
+    ## 106 Should filter in the Organizational Units tree
+    When I click on '[data-cy="generic-tree-node-00000000-0000-4000-8000-00000000eee1"]'
+    Then I expect current url is "{{ env.E2E_FRONT_URL }}/accounts?node=00000000-0000-4000-8000-00000000eee1"
+    And I expect the HTML element '[data-cy="organizational-unit-filter-input"]' to be visible
+    When I set the text "company" in the HTML element '[data-cy="organizational-unit-filter-input"]'
```
Comment (Zorin95670) [inline tests/features/front/AccountsPage.feature:119]: can you please break line before When

Comment (Hamilcar31) [reply tests/features/front/AccountsPage.feature:119]: Done

File: ui/tests/unit/components/tree/OrganizationalUnitsTree.spec.js [hunk 2]
```diff
@@ -232,6 +232,50 @@ describe('Test component: OrganizationalUnitsTree', () => {
     });
   });
 
+  describe('Test function: filterTreeNode', () => {
```
Comment (jcabannes) [inline ui/tests/unit/components/tree/OrganizationalUnitsTree.spec.js:235]: The unit tests cover casing and basic substring matching, but they don’t cover edge cases that can break at runtime (e.g., node.value.name missing/undefined, filter being null/undefined, or user input with leading/trailing whitespace). Add tests for these cases to lock in the intended behavior and prevent regressions.

File: ui/src/components/tree/OrganizationalUnitsTree.vue [hunk 3]
```diff
@@ -66,6 +68,19 @@ const route = useRoute();
...
+ * Filters a tree node based on a search string.
+ * @param node - Organizational unit tree node to evaluate.
+ * @param filter - Filter text entered by the user.
+ * @returns True if the node name matches the filter, otherwise false.
+ */
+function filterTreeNode(
+  node: TreeNode<OrganizationalUnitDTO>,
+  filter: string
+): boolean {
+  return node.value.name.toLowerCase().includes(filter.toLowerCase());
```
Comment (jcabannes) [inline ui/src/components/tree/OrganizationalUnitsTree.vue]: toLowerCase() will throw if node.value.name is ever null/undefined. Consider guarding with safe defaults (e.g., normalize name/filter to empty strings) before lowercasing.

Comment (Hamilcar31) [reply ui/src/components/tree/OrganizationalUnitsTree.vue]: Done

File: ui/tests/unit/components/tree/OrganizationalUnitsTree.spec.js (hunk 2 above)
Comment (Hamilcar31) [reply ui/tests/unit/components/tree/OrganizationalUnitsTree.spec.js:235]: Done

File: ui/tests/unit/components/tree/OrganizationalUnitsTree.spec.js [hunk 4]
```diff
@@ -232,6 +232,94 @@ describe('Test component: OrganizationalUnitsTree', () => {
...
+      expect(
+        wrapper.vm.filterTreeNode(makeNode('Engineering'), '  Engineering  ')
+      ).toBe(true);
+    });
+
+    it('should return true when the filter has only whitespace (treated as empty after trimming)', () => {
+      expect(wrapper.vm.filterTreeNode(makeNode('Engineering'), '   ')).toBe(
+        true
+      );
+    });
```
Comment (jcabannes) [inline ui/tests/unit/components/tree/OrganizationalUnitsTree.spec.js]: I am not sure that these two tests should return false, WDYT @Zorin95670 ?

File: ui/src/components/tree/OrganizationalUnitsTree.vue [hunk 5]
```diff
@@ -66,6 +68,21 @@ const route = useRoute();
...
+ * @param node - Organizational unit tree node to evaluate.
+ * @param filter - Filter text entered by the user.
+ * @returns True if the node name matches the filter, otherwise false.
+ */
+function filterTreeNode(
+  node: TreeNode<OrganizationalUnitDTO>,
+  filter: string
+): boolean {
+  const name = node.value?.name ?? '';
+  const filterStr = (filter ?? '').trim();
```
Comment (jcabannes) [inline ui/src/components/tree/OrganizationalUnitsTree.vue]: For me you should not trim filter value, WDYT @Zorin95670 ?

Comment (Zorin95670) [reply ui/src/components/tree/OrganizationalUnitsTree.vue]: See with you, split by ' ' and search with or for each index

File: ui/tests/unit/components/tree/OrganizationalUnitsTree.spec.js (hunk 4 above)
Comment (Hamilcar31) [reply ui/tests/unit/components/tree/OrganizationalUnitsTree.spec.js]: Done

File: ui/src/components/tree/OrganizationalUnitsTree.vue (hunk 5 above)
Comment (Hamilcar31) [reply ui/src/components/tree/OrganizationalUnitsTree.vue]: Done

File: ui/src/components/tree/OrganizationalUnitsTree.vue [hunk 6]
```diff
@@ -66,6 +68,24 @@ const route = useRoute();
...
+ */
+function filterTreeNode(
+  node: TreeNode<OrganizationalUnitDTO>,
+  filter: string
+): boolean {
+  const name = node.value?.name ?? '';
+  const filters = (filter ?? '')
+    .trim()
+    .split(' ')
+    .map((f) => f.trim());
```
Comment (Zorin95670) [inline ui/src/components/tree/OrganizationalUnitsTree.vue]: 'test     tata  '.split(' ').filter(f => f && f.length > 0).map((f) => f)

**Review by jcabannes — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]
